### DISCUSSION TITLE
Dynamic modals (data-binding params)

### DISCUSSION THREAD

**Discussion by DoneItInnovations** [livewire/flux#381] [Feature Requests]
It would be great if we could pass params (like id, or any other variable) with the modal's trigger button. 
It would pass the param into the dispatch event's params part. 

I mean it something like this:
```
<flux:modal.trigger name="edit-profile" :param="$row->id">
    <flux:button>Edit profile</flux:button>
</flux:modal.trigger>
```

**Comment by joesmalec** [comment]
I 100% need this

**Comment by ju5t** [comment]
What's the use case?

**Reply by SeraphixMD** [reply]
Row actions in a table would be my use-case.

**Reply by ju5t** [reply]
Yea but what are you trying to do? Do you want to reuse a modal?

**Reply by SeraphixMD** [reply]
That'd be a good example, yes. Reusing modals is already possible. However, as far as I was able to figure it out, I still need to make a round trip to the server with the ID of the entry, in between the user pressing the delete button, and pressing confirm in the modal that pops up, to make sure the server knows which entry to remove. Maybe I could use Alpine to store this ID temporarily on the client, haven't tried that yet, but it would make more sense to me, logically, if I could just pass that ID as an attribute to the modal and be done with it.

**Reply by richardvil** [reply]
@ju5t reuse is an option but the best example for this feature is with confirmation modal where wire:confirmation isn't enough. In those scenarios it's very common pass an id or some other parameters. So after the user confirm the action you can call a method from your component with wire:click for example.

**Reply by igmltd** [reply]
I just assign the id to a global alpine variable when the user clicks the button, then call the action with alpine e.g x-on:click=$wire.delete(AlpineVariableWithUserId)


```
<div x-data="{ userid : '' }">

// inside the loop

 <flux:modal.trigger name="delete-profile">
<flux:button x-on:click="userid = {{ $user->id }}" icon="trash"></flux:button>
</flux:modal.trigger>


// outside the loop

  <flux:modal name="delete-profile" class="min-w-[22rem] space-y-6">
        <div>
            <flux:heading size="lg">Delete project?</flux:heading>

            <flux:subheading>
                <p>You're about to delete this project.</p>
                <p>This action cannot be reversed.</p>
            </flux:subheading>
        </div>

        <div class="flex gap-2">
            <flux:spacer />

            <flux:modal.close>
                <flux:button variant="ghost">Cancel</flux:button>
            </flux:modal.close>

            <flux:button type="button" x-on:click=$wire.delete(userid)  variant="danger">Delete project</flux:button>
        </div>
    </flux:modal>
    
     </div>
```

**Reply by cord** [reply]
rich quickview on larger tables

**Reply by nigelhv** [reply]
The Flux modal would make a great replacement for a standard "show" page, but it needs to be easy to pass models, graphics, etc.

**Comment by mauritskorse** [comment]
I guess livewire v4 solves this by being able to pass slots into livewire components (https://livewire.laravel.com/docs/4.x/nesting#slots). So you will need to wrap your flux:modal into a livewire component to make this work.

This example also uses attribute forwarding:

```blade
<flux:modal.trigger name="viewPage">view post</flux:modal.tigger>
<livewire:modal name="viewPage" :post="$post">
    <flux:button>Close</flux:button>
</livewire:modal>
```

and your livewire component
```blade
<?php
use Livewire\Component;
new class extends Component{
  public Post $post;
};
?>
<flux:modal name="{{ $attributes->get('name') }}">
  <flux:heading>{{ $post->title }}</flux:heading>

   {{ $slot }}
</flux:modal>
```

**Comment by matz0r** [comment]
I do it like this:
```
  <flux:modal.trigger name="modal-name" wire:click="contentModal({{$something->id}})">
      show modal
  </flux:modal.trigger>
```

```
    <flux:modal name="modal-name">
        <div wire:text="content">
            my spinner component
        </div>
    </flux:modal>
```

and in the server-side component i just set content for that specific item that i want
```
    public function contentModal(int $someId): void
    {
        $this->content = "some data for $someId";
    }
```
